src/treelearner/feature_histogram.hpp

```diff
@@ -211,30 +213,45 @@ class FeatureHistogram {
 
 private:
   /*!
-  * \brief Calculate the split gain based on sum_gradients and sum_hessians
+  * \brief Calculate the split gain based on regularized sum_gradients and sum_hessians
   * \param sum_gradients
   * \param sum_hessians
   * \return split gain
   */
   double GetLeafSplitGain(double sum_gradients, double sum_hessians) const {
-    return (sum_gradients * sum_gradients) / (sum_hessians);
+    double abs_sum_gradients = fabs(sum_gradients);
```

Comment:
Done. Note that all of this is defined in `<cmath>`, which is not explicitly included in `feature_histogram.hpp`, but included through `#include "split_info.hpp"`. This doesn't seem to be handled in a systematic way in this project, so do you have any preference on how to handle this?